runners/spark/src/main/java/org/apache/beam/runners/spark/translation/MultiDoFnFunction.java

```diff
@@ -61,6 +66,17 @@
 public class MultiDoFnFunction<InputT, OutputT>
     implements PairFlatMapFunction<Iterator<WindowedValue<InputT>>, TupleTag<?>, WindowedValue<?>> {
 
+  private static final Logger LOG = LoggerFactory.getLogger(MultiDoFnFunction.class);
+
+  /** JVM wide side input cache. */
+  private static final Map<String, CachedSideInputReader> sideInputReaders =
+      Collections.synchronizedMap(new WeakHashMap<>());
+
+  /**
+   * Id that is consistent among executors. We can not use stepName because of possible collisions.
```

Comment:
I do not get meaning of 'consistent' here. Do you mean random even within one JVM ?

Reply:
After deserialization on the executor side

Reply:
ok, I see.